### ISSUE TITLE
Change in path parsing in URI.parse()

### ISSUE THREAD

**Issue by jonleighton** [elixir-lang/elixir#11363]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 1.13.0-rc.0
* Operating system: Linux

### Current behavior (1.13.0-rc.0)

```
iex(1)> URI.parse("https://foo.com")
%URI{
  authority: "foo.com",
  fragment: nil,
  host: "foo.com",
  path: "",
  port: 443,
  query: nil,
  scheme: "https",
  userinfo: nil
}
```

### Expected behavior (1.12.3)

```
iex(1)> URI.parse("https://foo.com")
%URI{
  authority: "foo.com",
  fragment: nil,
  host: "foo.com",
  path: nil,
  port: 443,
  query: nil,
  scheme: "https",
  userinfo: nil
}
```

The `path` field is `nil` on 1.12, but `""` on 1.13.

---

To help others who may be searching this problem: I discovered this via the following error from [Redix](https://github.com/whatyouhide/redix):

```
    ▏ ▏ ▏ ▏ ▏ ** (FunctionClauseError) no function clause matching in Redix.URI.database/1
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (redix 1.0.0) lib/redix/uri.ex:36: Redix.URI.database(%URI{authority: "localhost:16379", fragment: nil, host: "localhost", path: "", port: 16379, query: nil, scheme: "redis", userinfo: nil})
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (redix 1.0.0) lib/redix/uri.ex:16: Redix.URI.opts_from_uri/1
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (redix 1.0.0) lib/redix.ex:318: Redix.start_link/2
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (stdlib 3.15.2) supervisor.erl:414: :supervisor.do_start_child_i/3
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (stdlib 3.15.2) supervisor.erl:400: :supervisor.do_start_child/2
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (stdlib 3.15.2) supervisor.erl:384: anonymous fn/3 in :supervisor.start_children/2
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (stdlib 3.15.2) supervisor.erl:1234: :supervisor.children_map/4
    ▏ ▏ ▏ ▏ ▏ ▏ ▏ (stdlib 3.15.2) supervisor.erl:350: :supervisor.init_children/2
```

I was able to work around it by changing my `REDIS_URL` from:

```
REDIS_URL=redis://localhost:16379
```

to:

```
REDIS_URL=redis://localhost:16379/
```

**Comment by lukaszsamson** [comment]
Will the fix be backported to 1.13? It breaks one of the tests in elixir-LS
```
  2) test path_from_uri no `path` in URI (ElixirLS.LanguageServer.SourceFileTest)
     test/source_file_test.exs:700
     Assertion with == failed
     code:  assert path == "/"
     left:  ""
     right: "/"
     stacktrace:
       test/source_file_test.exs:706: (test)
```

**Comment by josevalim** [comment]
Done!
